packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx

```diff
@@ -177,260 +178,274 @@ const fillRequiredFields = async () => {
 
-  it('renders and saves registration obs', async () => {
-    const user = userEvent.setup();
-
-    mockedSaveEncounter.mockResolvedValue({});
-    mockedUseConfig.mockReturnValue(configWithObs);
-
-    render(
-      <ResourcesContext.Provider value={mockResourcesContextValue}>
-        <Router>
-          <PatientRegistration isOffline={false} savePatientForm={FormManager.savePatientFormOnline} />
-        </Router>
-      </ResourcesContext.Provider>,
-    );
-
-    await fillRequiredFields();
-    const customSection = screen.getByLabelText('Custom Section');
-    const weight = within(customSection).getByLabelText('Weight (kg)');
-    await user.type(weight, '50');
-    const complaint = within(customSection).getByLabelText('Chief Complaint');
-    await user.type(complaint, 'sad');
-    const nationality = within(customSection).getByLabelText('Nationality');
-    await user.selectOptions(nationality, 'USA');
-
-    await user.click(screen.getByText('Register Patient'));
-
-    await waitFor(() => expect(mockedSavePatient).toHaveBeenCalled());
-    await waitFor(() =>
-      expect(mockedSaveEncounter).toHaveBeenCalledWith(
-        expect.anything(),
-        expect.objectContaining<Partial<Encounter>>({
-          encounterType: 'reg-enc-uuid',
-          patient: 'new-pt-uuid',
-          obs: [
-            { concept: 'weight-uuid', value: 50 },
-            { concept: 'chief-complaint-uuid', value: 'sad' },
-            { concept: 'nationality-uuid', value: 'usa' },
-          ],
-        }),
-      ),
-    );
-  });
+      mockedUseConfig.mockReturnValue(configWithObs);
+
+      render(
+        <ResourcesContext.Provider value={mockResourcesContextValue}>
+          <Router>
+            <PatientRegistration isOffline={false} savePatientForm={FormManager.savePatientFormOnline} />
+          </Router>
+        </ResourcesContext.Provider>,
+      );
+
+      await fillRequiredFields();
+      const customSection = screen.getByLabelText('Custom Section');
+      const weight = within(customSection).getByLabelText('Weight (kg) (optional)');
+      await user.type(weight, '-999');
+
+      mockedSaveEncounter.mockRejectedValue({ status: 400, responseBody: { error: { message: 'an error message' } } });
+
+      await user.click(screen.getByText('Register Patient'));
+
+      await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(1));
+      await waitFor(() => expect(mockedSaveEncounter).toHaveBeenCalledTimes(1));
+      await waitFor(() =>
+        expect(mockedShowToast).toHaveBeenCalledWith(expect.objectContaining({ description: 'an error message' })),
+      );
 
-  it('retries saving registration obs after a failed attempt', async () => {
-    const user = userEvent.setup();
+      mockedSaveEncounter.mockResolvedValue({});
 
-    mockedUseConfig.mockReturnValue(configWithObs);
+      await user.click(screen.getByText('Register Patient'));
+      await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(2));
```

Comment:
Why this needs to be called two times?

Reply:
Because it is retrying after the failed attempt.